spring-xd-messagebus-spi/src/main/java/org/springframework/xd/dirt/integration/bus/MessageBusSupport.java

```diff
@@ -1103,4 +1104,19 @@ public void handleMessage(Message<?> message) throws MessagingException {
 
 	}
 
+	/**
+	 * No op method that can be implemented by specific message bus to store message headers to acknowledge
+	 * the messages manually later.
+	 *
+	 * @param messageHeaders the message headers
+	 */
+	public void storeForManualAck(MessageHeaders messageHeaders) {
+	}
+
+	/**
+	 * Perform manual acknowledgement based on the metadata stored in message bus.
+	 */
+	public void doManualAck() {
+	}
+
```

Comment:
Should these eventually be pulled up to `MessageBus` ? (Obviously not for 1.1.x)

Also, please move them above the `DefaultPartitionSelectorClass`; convention is to have classes at the end of the file.